### PR TITLE
gh #15 Code Cleanup in L1 for power_manager

### PR DESCRIPTION
**Pull request by Sumip98**
1. Clean up unnecessary code in L1:
   Removed the unused variables, and macros and compiled with zero warnings.
   FATAL used in all tests is removed except for init and term APIs.  
   Reviewed the usage of casting.
   
 2. Re-checked if any new API interface has been added recently and requires an L1 test case, no new API interface was found.

### CODE REVIEW

File: src/test_l1_plat_power.c [hunk 1]
```diff
@@ -71,7 +71,7 @@
 #include <ut.h>
 #include <ut_log.h>
 #include "ut_kvp_profile.h"
-
+
```
Comment (thanushreevani) [inline src/test_l1_plat_power.c:74]: remove extra spaces

File: src/test_l1_plat_power.c [hunk 2]
```diff
@@ -115,11 +115,11 @@ void test_l1_plat_power_positive_PLAT_INIT (void)
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID); 
     pmStatus_t ret;
 
-    // Variation/Step 01: Call PLAT_INIT() - open interface
+    // Step 01: Call PLAT_INIT() - open interface
     ret = PLAT_INIT();
     UT_ASSERT_EQUAL_FATAL(ret, PWRMGR_SUCCESS); // Ensure the returned value is PWRMGR_SUCCESS
```
Comment (thanushreevani) [inline src/test_l1_plat_power.c:120]: comment added like this can be removed in whole file

File: src/test_l1_plat_power.c [hunk 3]
```diff
@@ -284,22 +283,23 @@ void test_l1_plat_power_positive_PLAT_API_SetPowerState (void)
...
 
-    // Variation/step 02:Call PLAT_API_SetPowerState() -Iterate through various power states fetched from the profile file
+    // step 02:Call PLAT_API_SetPowerState() -Iterate through various power states fetched from the profile file
     for(int i = 0; i < countOfPowerState; i++) {
         snprintf(keyvalue, POWER_MANAGER_KEY_SIZE, "powermanager/PowerStates/%d", i);
         powerState = UT_KVP_PROFILE_GET_UINT32(keyvalue);
```
Comment (thanushreevani) [inline src/test_l1_plat_power.c:293]: Review the usage of casting:
Eg: from eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
to eIndicator =(dsFPDIndicator_t) UT_KVP_PROFILE_GET_UINT32(buffer)

Check everywhere in document where ever you are doing GET from yaml file

File: src/test_l1_plat_power.c (hunk 1 above)
Comment (Sumip98) [reply src/test_l1_plat_power.c:74]: Done

File: src/test_l1_plat_power.c (hunk 2 above)
Comment (Sumip98) [reply src/test_l1_plat_power.c]: Done

File: src/test_l1_plat_power.c (hunk 3 above)
Comment (Sumip98) [reply src/test_l1_plat_power.c]: Done

File: src/test_l1_plat_power.c [hunk 4]
```diff
@@ -71,12 +71,12 @@
 #include <ut.h>
 #include <ut_log.h>
 #include "ut_kvp_profile.h"
-
+
```
Comment (thanushreevani) [inline src/test_l1_plat_power.c:74]: space there here still

File: src/test_l1_plat_power.c [hunk 5]
```diff
@@ -489,21 +486,21 @@ void test_l1_plat_power_positive_PLAT_API_SetWakeupSrc (void)
...
-    countOfWakeupSources = UT_KVP_PROFILE_GET_UINT32("powermanager/CountOfWakeupSources");
+    // Step 02:Loop through all wakeup sources and set to false
+    countOfWakeupSources = UT_KVP_PROFILE_GET_UINT8("powermanager/CountOfWakeupSources");
     UT_LOG_DEBUG ("Count Of Wakeup Sources %d" , countOfWakeupSources);
-    
+
     for(int i = 0; i < countOfWakeupSources; i++){
         snprintf(keyvalue, POWER_MANAGER_KEY_SIZE, "powermanager/WakeupSources/%d", i);
-        srcType = UT_KVP_PROFILE_GET_UINT32(keyvalue);
+        srcType =(PWRMgr_PowerState_t)UT_KVP_PROFILE_GET_UINT32(keyvalue);
```
Comment (thanushreevani) [inline src/test_l1_plat_power.c:499]: this should be of type PWRMGR_WakeupSrcType_t ! please check everywhere correctly

Comment (Sumip98) [reply src/test_l1_plat_power.c]: Done

File: src/test_l1_plat_power.c (hunk 4 above)
Comment (Sumip98) [reply src/test_l1_plat_power.c]: Done

Comment (thanushreevani) [reply src/test_l1_plat_power.c]: check this

Comment (Sumip98) [reply src/test_l1_plat_power.c]: Done

File: src/test_l1_plat_power.c [hunk 6]
```diff
@@ -623,13 +621,13 @@ void test_l1_plat_power_positive_PLAT_API_GetWakeupSrc (void)
...
 
-    //Step 03:Getting the supported wakeup sources
+    //Step 03:Getting and setting supported wakeup sources, then verifying their states
     for(int i = 0; i < countOfWakeupSources; i++){
         snprintf(keyvalue, POWER_MANAGER_KEY_SIZE, "powermanager/WakeupSources/%d", i);
-        srcType = UT_KVP_PROFILE_GET_UINT32(keyvalue);
+        srcType =(PWRMGR_WakeupSrcType_t)UT_KVP_PROFILE_GET_UINT32(keyvalue);
         UT_LOG_DEBUG("Invoking PLAT_API_SetWakeupSrc with srcType: %d and enable: true", srcType);
 
         ret = PLAT_API_SetWakeupSrc(srcType, true);
```
Comment (bhanucbp) [inline src/test_l1_plat_power.c:633]: remove continue in the loop

**Review by bhanucbp — COMMENTED** [review]
address the review comment

**Review by bhanucbp — APPROVED** [review]
ship it

**Review by thanushreevani — APPROVED** [review]

**Pull request merged**